ipn/ipnlocal/local.go

```diff
@@ -1881,6 +1881,12 @@ func magicDNSRootDomains(nm *netmap.NetworkMap) []dnsname.FQDN {
 			// TODO: propagate error
 			return nil
 		}
+		if strings.HasSuffix(fqdn.WithoutTrailingDot(), ".beta.tailscale.net") {
```

Comment:
hard-coding tailscale-specific stuff in here? 😬

Reply:
I don't feel super bad about hardcoding tailscale-specific stuff into the tailscale codebase :)

Reply:
Especially so because the entirety of the MagicDNS configuration is generated on the client side, the only thing control says is "do a magic dns", so... yeah, we're doing a magic dns.